app/src/main/java/org/mifos/mobile/ui/home/HomeOldFragment.kt

```diff
@@ -215,38 +255,36 @@ class HomeOldFragment : BaseFragment(), OnRefreshListener {
      */
     fun showUserDetails(client: Client?) {
         this.client = client
-        binding.tvUserName.text = getString(R.string.hello_client, client?.displayName)
     }
 
     /**
      * Provides with Client image fetched from server
      *
      * @param bitmap Client Image
      */
-    fun showUserImage(bitmap: Bitmap?) {
-        activity?.runOnUiThread {
-            if (bitmap != null) {
-                binding.ivCircularUserImage.visibility = View.VISIBLE
-                binding.ivCircularUserImage.setImageBitmap(bitmap)
-            } else {
-                val userName = preferencesHelper?.clientName.let { savedName ->
-                    if (savedName.isNullOrBlank()) {
-                        getString(R.string.app_name)
-                    } else {
-                        savedName
-                    }
+    private fun showUserImage(bitmap: Bitmap?) {
+        userBitmap = if (bitmap != null) {
+            bitmap
+        } else {
+            val userName = preferencesHelper?.clientName.let { savedName ->
+                if (savedName.isNullOrBlank()) {
+                    getString(R.string.app_name)
+                } else {
+                    savedName
                 }
-
-                val drawable = TextDrawable.builder()
-                    .beginConfig()
-                    .toUpperCase()
-                    .endConfig()
-                    .buildRound(
-                        userName.substring(0, 1),
-                        requireContext().getThemeAttributeColor(R.attr.colorPrimary),
-                    )
-                binding.ivCircularUserImage.setImageDrawable(drawable)
             }
+
+            val textDrawable = TextDrawable.builder()
```

Comment:
Remove this and build a equivalent compose component and move in HomeScreen and manage the data in `HomeUiState`